src/cosalib/meta.py

```diff
@@ -138,7 +138,8 @@ def read(self):
         # detect conflicts
         with open(self.path) as f:
             self._initial_timestamp = os.fstat(f.fileno()).st_mtime
-            self.update(json.load(f))
+        # Read under a lock to prevent race conditions
+        self.update(load_json(self.path))
```

Comment:
This makes `self._initial_timestamp` in turn possibly racy. It seems then like `load_json` should return both the JSON and the timestamp (or maybe better make that a new function like `load_json_with_ts` and have `load_json` use that).

OTOH, it doesn't seem like `_initial_timestamp` is actually used anywhere yet, so... maybe instead we just nuke lines 137-140?

Reply:
Okay, now, I think we should be less-racy. Instead of creating an arbitrary meta-stamp, let's build it into the base `meta.json`.